src/hexEditorProvider.ts

```diff
@@ -364,10 +374,27 @@ export class HexEditorProvider implements vscode.CustomEditorProvider<HexDocumen
 						.map(s => document.readBuffer(s[0], s[1] - s[0])),
 				);
 				const flatParts = flattenBuffers(parts);
-				const encoded = message.asText
-					? new TextDecoder().decode(flatParts)
-					: base64.fromUint8Array(flatParts);
-				vscode.env.clipboard.writeText(encoded);
+				const filenameWoutExt = ((path: string): string => {
+					const filename = path.split("/").pop()!;
+					return filename.substring(0, filename.lastIndexOf(".")) || filename;
+				})(document.uri.path);
+				const copyAsFormats: { [K in CopyFormat]: (buffer: Uint8Array) => void } = {
```

Comment:
I would make this object an export of `copyAs`, and then have the signature `(buffer: Uint8Array, filename: string) => void` so you can pass it in there. Then the HexEditorProvider doesn't have to care about the specifics of all the formats